src/main/java/com/endava/internship/warmup/domain/service/ArrayProcessorWithForLoops.java

```diff
@@ -111,7 +181,44 @@ public int findSecondMax(final int[] input) {
      */
     @Override
     public int[] mergeSortedArrays(int[] input, int[] input2) throws IllegalArgumentException {
-        throw new UnsupportedOperationException("Method not implemented yet");
+        final int N = input.length;
+        final int M = input2.length;
+        int[] newValues = new int[N + M];
+        int i = 0, j = 0, k = 0;
+
+        while(i < N && j < M) {
+            if(input[i] < input2[j]) {
+                if(i > 0 && input[i] < input[i - 1]) {
```

Comment:
remove all verifications like these from everywhere, add a single try-catch block, there in the try block will be all code, and in catch will be "throw new IllegalArgumentException()"